### DISCUSSION TITLE
Auto Implemented Properties with Actions

### DISCUSSION THREAD

**Discussion by hbraasch** [dotnet/csharplang#654] [General]
This is probably not a new suggestion, but other than just allowing 
e.g.  bool IsDirty {get; set;} = false;

Please consider allowing us to add code to them without losing the advantage of anonymous backing fields... 
e.g. bool IsDirty {get; set (value) => { MyMethod(value)};} = false;

We can now trigger events if properties change, without having to code with backing fields. This cleans up

**Comment by jnm2** [comment]
I'd much rather see what has been suggested already (https://github.com/dotnet/csharplang/issues/140), the `field` keyword:

```c#
bool IsDirty { get; set => field = MyMethod(value); } = false;
```

**Comment by hbraasch** [comment]
Should be fine, although one loses this in-line capability e.g.:
bool IsDirty { get; set => field = var a= DoThisFirst(); Final(value, a, b); } = false;
But I guess all this will have to fit inside MyMethod(). It's OK

**Comment by alrz** [comment]
That's quite useful, but when you have a readonly property my bet is on property-scoped fields,
```cs
object Foo { object field; get => field ??= Compute(); }
```

**Comment by hbraasch** [comment]
OK, I now realized I missed the meaning of the <<field>> keyword. It represents the <<anonymous backing field>>. Thanks for making that clear.

If the implementation can also allow multiple in-line statements , that would satisfy my requirement. I simply want to trigger actions when <<set>>ing without having to code the complete property. Maybe a better example based on the <<field>> keyword approach would be:
object Foo { get => field ??= Compute(); set => {field = value; FireCustomAction(value);} } = false;

**Comment by jnm2** [comment]
@hbraasch

````markdown
```c#
// Syntax-highlighted block
object Foo { get => field ??= Compute(); set => {field = value; FireCustomAction(value);} } = false;
```
````

Gives 

```c#
// Syntax-highlighted block
object Foo { get => field ??= Compute(); set => {field = value; FireCustomAction(value);} } = false;
```

**Comment by jnm2** [comment]
I do really want to see https://github.com/dotnet/csharplang/issues/140. It'll cover so many of the motivations in these asks.

**Comment by jnm2** [comment]
Using a backing field without naming it was considered in https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-01.md
